torch/csrc/jit/script/parser.h

```diff
@@ -24,9 +24,9 @@ struct Parser {
     parseOperatorArguments(inputs, attributes);
     return Apply::create(
         range,
-        ident,
-        List(range, std::move(inputs)),
-        List(range, std::move(attributes)));
+        Ident(ident),
+        List<Expr>(makeList(range, std::move(inputs))),
+        List<Attribute>(makeList(range, std::move(attributes))));
```

Comment:
This List<Attribute> node doesn't actually check anything on construction. This does not appear any safer than the code we had before.  When constructing a tree, we should be checking this. When consuming an already constructed tree, it is ok to just assume the tree was constructed correctly.

Reply:
Can we add a constructor `List<Attribute>::create(std::vector<TreeRef> &)` that does a checked construction of these lists? Then we can have `List<Attribute>(TreeRef &)` perform the unchecked construction as it already does.